src/main/java/com/google/devtools/build/lib/rules/cpp/LibrariesToLinkCollector.java

```diff
@@ -607,9 +607,9 @@ private void addDynamicInputLinkOptions(
     // -l:foo -> foo.so
     // -l:libfoo.so.1 -> libfoo.so.1
     boolean hasCompatibleName =
-        name.startsWith("lib") || (!name.endsWith(".so") && !name.endsWith(".dylib"));
+        name.startsWith("lib") || (!name.endsWith(".so") && !name.endsWith(".dylib") && !name.endsWith(".dll"));
```

Comment:
Not familiar with java and regex, but... If `hasCompatibleName` checks name compatibility, should it be:
`name.startsWith("lib") || (name.endsWith(".so") || name.endsWith(".dylib") || name.endsWith(".dll"));`  ?
Otherwise names `lib\<name\>.so` are stripped, but names `\<name\>.so` are not. At least in my case I noticed this behaviour.